presto-main/src/main/java/com/facebook/presto/metadata/FunctionNamespace.java

```diff
@@ -23,70 +23,28 @@
 import com.facebook.presto.sql.analyzer.TypeSignatureProvider;
 import com.facebook.presto.sql.tree.QualifiedName;
 
-import javax.annotation.concurrent.ThreadSafe;
-
 import java.util.List;
 
-import static java.util.Objects.requireNonNull;
-
-@ThreadSafe
-class FunctionNamespace
+public interface FunctionNamespace
 {
-    private final FunctionRegistry registry;
-
-    public FunctionNamespace(FunctionRegistry registry)
-    {
-        this.registry = requireNonNull(registry, "registry is null");
-    }
-
-    public void addFunctions(List<? extends SqlFunction> functions)
-    {
-        registry.addFunctions(functions);
-    }
+    void addFunctions(List<? extends SqlFunction> functions);
 
-    public List<SqlFunction> listFunctions()
-    {
-        return registry.list();
-    }
+    List<SqlFunction> listFunctions();
 
-    public FunctionHandle lookupFunction(QualifiedName name, List<TypeSignatureProvider> parameterTypes)
-    {
-        return registry.lookupFunction(name, parameterTypes);
-    }
+    FunctionHandle lookupFunction(QualifiedName name, List<TypeSignatureProvider> parameterTypes);
```

Comment:
It's not clear to me whether a function namespace in general would need both `lookupFunction` and `resolveFunction`. Conceptually they should just need a `resolveFunction`. `lookupFunction` is used for 1) testing; 2) function dependency in annotation; 3) optimization rules. It's not obvious that these would be useful for other function namespaces.